sled-agent/src/sled_agent.rs

```diff
@@ -808,7 +819,28 @@ impl SledAgent {
         &self,
         config: OmicronPhysicalDisksConfig,
     ) -> Result<DisksManagementResult, Error> {
-        Ok(self.storage().omicron_physical_disks_ensure(config).await?)
+        // First, tell the storage subsystem which disks should be managed.
```

Comment:
This is the crux is this PR:

- We update the set of disks. This should occur when Nexus triggers disk expungement -- the list of physical disks in `config` will include one less U.2.
- AFTER this happens, the sled agent can then await all internal uses of U.2s, and ensure that the view of "what disks should we use" is at least as high as a generation *after* expungement.